crowd_evacuation/civilian_agent.py

```diff
@@ -159,10 +203,12 @@ def _interact(self, other):
             self._known_exits = shared_known_exits
             other._known_exits = shared_known_exits
 
-    # _looking_around: an agents look around in visible range(5X5) and find out other agents.
     def _looking_around(self):
+        """
+        :return: surrounding agents, possible moving positions and obstacles
+        """
         # the list of objects surrounding an agent, 5x5 range, exclude the center where the agent is standing
-        surrounding_agents = self.model.grid.get_neighbors(self.pos, moore=True, include_center=False, radius=5)
+        surrounding_agents = self.model.grid.get_neighbors(self.pos, moore=True, include_center=False, radius=self._speed+2)
```

Comment:
why do slower people see less? :)
Or I misunderstand this.

Reply:
What i did was that slower people are also the older one. However, at this moment age also varies slightly with weight. So I thought that the most logical way is to first create a visual range which varies with age and then calculate the speed based on the visual range and the weight, which adds a negative penalty to the visual range (which is based on age)